### PR TITLE
Implement Iteration 2 and 3 to main branch.

### PR DESCRIPTION
**Pull request by ConnorAndersonLarson**
## This will add the following features to our main branch:  
- Allow for switching between pages based on button presses. 
- Allow for creation of custom book covers.
- Storing custom made covers to be used in random cover creation.
## Testing can take place by opening index.html in terminal and testing out the various buttons. Note the 'Saved Covers' page is blank for this current iteration.  

Book Creation
Change Pages

### CODE REVIEW

**Review by itsnameissteven — COMMENTED** [review]
The changes look good.  Refactored code is a lot cleaner. +++ 1 on the gif. Everything looks real smooth. I think eventually we will want to consider changing some variable and function names and taking out the commented out code.

File: src/main.js [hunk 1]
```diff
@@ -59,8 +49,7 @@ function randomCover(){
     var randomDescOne = descriptors[getRandomIndex(descriptors)]
     var randomCover = covers[getRandomIndex(covers)]
     var randomTitle = titles[getRandomIndex(titles)]
-   currentCover([randomTitle, randomCover, randomDescOne, randomDescTwo])
-   //savedCovers.push(randCover);
+   makeCurrentCover([randomTitle, randomCover, randomDescOne, randomDescTwo])
```
Comment (itsnameissteven) [inline src/main.js]: I would suggest switching the first two elements in this array to make it consistent with the order of Cover class.  We'll also have to edit the makeCurrentCover function accordingly

File: src/main.js [hunk 2]
```diff
@@ -18,31 +17,22 @@ var titleInput = document.querySelector('.user-title');
...
-// Create your event handlers and other functions here 👇
-
-
-// We've provided one function to get you started
 function getRandomIndex(array) {
   return Math.floor(Math.random() * array.length);
 }
 
-function currentCover(array) {
+function makeCurrentCover(array) {
```
Comment (itsnameissteven) [inline src/main.js:35]: If we edit the array below change this one

File: src/main.js [hunk 3]
```diff
@@ -59,45 +49,49 @@ function randomCover(){
...
     var randomCover = covers[getRandomIndex(covers)]
     var randomTitle = titles[getRandomIndex(titles)]
-   currentCover([randomTitle, randomCover, randomDescOne, randomDescTwo])
-   //savedCovers.push(randCover);
+   makeCurrentCover([randomTitle, randomCover, randomDescOne, randomDescTwo])
 }
 randomCover()
 
 function changeToSavedCovers() {
-  homeButton.classList.remove('hidden');
```
Comment (gaj23) [inline src/main.js:68]: Are you able to use .toggle instead of add/remove in any places?

File: src/main.js [hunk 4]
```diff
@@ -59,45 +49,49 @@ function randomCover(){
...
   saveCoverButton.classList.remove('hidden')
   homeButton.classList.add('hidden');
   randomCoverButton.classList.remove('hidden')
   mainCover.classList.remove('hidden')
-  savedCoversSection.classList.add('hidden');
-  formView.classList.add('hidden');
 };
 
-function makeBookCover() {
+function changePage(num){
```
Comment (gaj23) [inline src/main.js:74]: Nice This is a smart use of a helper function.

Comment (gaj23) [reply src/main.js:74]: I am curious as to why toggle wasn't utilized, though. But I like the work around.

**Review by gaj23 — COMMENTED** [review]
I really liked your helper function changePage. Very fun. 

I would say your branch naming is a little confusing. This branch contains both iteration 2 and 3? If so, maybe ask where you could have merged iteration 2 sooner just in case something broke in iteration 3. It's always good to have something you can go back to just in case.

File: src/main.js (hunk 4 above)
Comment (ConnorAndersonLarson) [reply src/main.js:74]: We originally were using an entirely different method to hide/unhide everything and more recently converted to classList.Add/Remove. The syntax of that earlier method still lives on in our code (and I can't quite think through how to toggle properly between three different buttons).

File: src/main.js (hunk 1 above)
Comment (ConnorAndersonLarson) [reply src/main.js]: Went and edited per your recommendations. Take a look and see what you think!

File: src/main.js (hunk 4 above)
Comment (itsnameissteven) [reply src/main.js:74]: Yes I was having the same issue when working with .toggle()

File: src/main.js (hunk 2 above)
Comment (ConnorAndersonLarson) [reply src/main.js:35]: Done

**Pull request merged**